modules/images/webp-uploads/load.php

```diff
@@ -340,3 +340,130 @@ function webp_uploads_wp_get_missing_image_subsizes( $missing_sizes, $image_meta
 }
 
 add_filter( 'wp_get_missing_image_subsizes', 'webp_uploads_wp_get_missing_image_subsizes', 10, 3 );
+
+/**
+ * Filters `the_content` to update images so that they use the preferred MIME type where possible.
+ *
+ * By default, this is `image/webp`, if the current attachment contains the targeted MIME
+ * type. In the near future this will be filterable.
+ *
+ * Note that most of this function will not be needed for an eventual core implementation as it
+ * would rely on `wp_filter_content_tags()`.
+ *
+ * @since n.e.x.t
+ *
+ * @see wp_filter_content_tags()
+ *
+ * @param string $content The content of the current post.
+ * @return string The content with the updated references to the images.
+ */
+function webp_uploads_update_image_references( $content ) {
+	// This content does not have any tag on it, move forward.
+	if ( ! preg_match_all( '/<(img)\s[^>]+>/', $content, $img_tags, PREG_SET_ORDER ) ) {
+		return $content;
+	}
+
+	$images = array();
+	foreach ( $img_tags as list( $img ) ) {
+		// Find the ID of each image by the class.
+		if ( ! preg_match( '/wp-image-([\d]+)/i', $img, $class_name ) ) {
+			continue;
+		}
+
+		if ( empty( $class_name ) ) {
+			continue;
+		}
+
+		// Make sure we use the last item on the list of matches.
+		$attachment_id = (int) end( $class_name );
+
+		if ( ! $attachment_id ) {
+			continue;
+		}
+
+		// Create an array if attachment_id has not been added to the images array.
+		if ( empty( $images[ $attachment_id ] ) ) {
+			$images[ $attachment_id ] = array();
+		}
+
+		$images[ $attachment_id ][] = $img;
+	}
+
+	foreach ( $images as $attachment_id => $images_tag ) {
+		$metadata = wp_get_attachment_metadata( $attachment_id );
+
+		foreach ( $images_tag as $img ) {
+			$content = str_replace( $img, webp_uploads_img_tag_update_mime_type( $img, $metadata ), $content );
+		}
+	}
+
+	return $content;
+}
+
+/**
+ * Finds all the urls with *.jpg and *.jpeg extension and updates with *.webp version for the provided image
+ * for the specified image sizes, the *.webp references are stored inside of each size.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $image An <img> tag where the urls would be updated.
+ * @param array  $metadata An associative array with the metadata for the image.
+ * @return string The updated img tag.
+ */
+function webp_uploads_img_tag_update_mime_type( $image, array $metadata ) {
```

Comment:
While I see your point that it's technically not necessary, I think it makes sense to align this function already to be how it can easily be integrated into WordPress core, i.e. we could use the same parameters `$image, $context, $attachment_id`.

Currently you're calling `wp_get_attachment_metadata()` outside of this function, but that doesn't really have any performance benefit. Even if an attachment is present 10 times, these additional calls will all use the same data that is being cached, even on a system without object cache it will only be a single DB request.

Especially with the `_prime_post_caches()` call, there should be no concern of moving the `wp_get_attachment_metadata()` call into this function, which then aligns it with other core functions (which also call `wp_get_attachment_metadata()` for every image, since that's not really a problem).